src/Sparrow/LowMemory/MemoryInformation.cs

```diff
@@ -80,8 +81,21 @@ public struct MemoryStatusEx
 
         public static bool EnableEarlyOutOfMemoryChecks = false; // we don't want this to run on the clients
 
+        [ThreadStatic]
+        private static bool _disableOutOfMemoryChecks;
+
+        public static IDisposable DisableOutOfMemoryChecks()
+        {
+            _disableOutOfMemoryChecks = true;
+
+            return new DisposableAction(() => _disableOutOfMemoryChecks = false);
```

Comment:
In Voron we typically create dedicated struct which is disposable, no need the use DisposableAction and move to to Sparrow e.g. `public struct ExitWriteLock : IDisposable`